library/src/main/java/com/pokegoapi/util/Signature.java

```diff
@@ -38,15 +36,33 @@ public static void setSignature(PokemonGo api, RequestEnvelopeOuterClass.Request
 		SignatureOuterClass.Signature.Builder sigBuilder = SignatureOuterClass.Signature.newBuilder()
 				.setLocationHash1(getLocationHash1(api, authTicketBA, builder))
 				.setLocationHash2(getLocationHash2(api, builder))
-				.setUnk22(ByteString.copyFrom(uk22))
+				.setUnknown22(ByteString.copyFrom(uk22))
 				.setTimestamp(api.currentTimeMillis())
-				.setTimestampSinceStart(curTime - api.startTime);
+				.setTimestampSinceStart(curTime - api.startTime)
+				.setActivityStatus(
```

Comment:
I added the ActivityStatus, because the official app is sending all values inside the ActivityStatus setted to 1.